solarforecastarbiter/metrics/calculator.py

```diff
@@ -154,6 +157,12 @@ def calculate_deterministic_metrics(processed_fx_obs, categories, metrics,
                     'observation': obs,
                     'reference': ref_fx}, axis=1)
 
+    # Force `groupby` to be consistent with `interval_label`, i.e., if
+    # `interval_label == ending`, then the last interval should be in the bin
+    closed = datamodel.CLOSED_MAPPING[processed_fx_obs.interval_label]
+    if closed == "ending":
+        df.index -= pd.Timedelta("1ns")
```

Comment:
codecov says this line isn't hit. Looks like it will be if you fix the bug above.